src/elements/Input2/Input2.tsx

```diff
@@ -0,0 +1,570 @@
+import { EventEmitter } from "events"
+
+import { THEME } from "@artsy/palette-tokens"
+import themeGet from "@styled-system/theme-get"
+import {
+  RefObject,
+  forwardRef,
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useMemo,
+  useRef,
+  useState,
+} from "react"
+import {
+  LayoutAnimation,
+  NativeSyntheticEvent,
+  TextInput,
+  TextInputFocusEventData,
+  TextInputProps,
+  TouchableOpacity,
+  ViewProps,
+} from "react-native"
+import Animated, { useAnimatedStyle, useSharedValue, withTiming } from "react-native-reanimated"
+import styled from "styled-components"
+import { INPUT_VARIANTS, InputState, InputVariant, getInputState, getInputVariant } from "./helpers"
+import { EyeClosedIcon, EyeOpenedIcon, TriangleDown, XCircleIcon } from "../../svgs"
+import { useColor, useSpace } from "../../utils/hooks"
+import { useMeasure } from "../../utils/hooks/useMeasure"
+import { Flex } from "../Flex"
+import { Spinner } from "../Spinner"
+import { Text } from "../Text"
+import { Touchable } from "../Touchable"
+
+export const input2Events = new EventEmitter()
+
+export const emitInput2ClearEvent = () => {
+  input2Events.emit("clear")
+}
+
+export interface Input2Props extends TextInputProps {
+  addClearListener?: boolean
+  // We are applying some optimisations to make sure the UX is smooth
+  // These lead to some issues when the parent component wants further control of the value
+  disableOnChangeOptimisation?: boolean
```

Comment:
To see the comment in VSCode.

<img width="945" alt="Screenshot 2024-03-27 at 14 12 19" src="https://github.com/artsy/palette-mobile/assets/4691889/5f9a1874-9bea-40a0-9211-f0ca46670549">


```suggestion
  /**
   * We are applying some optimisations to make sure the UX is smooth
   * These lead to some issues when the parent component wants further control of the value
   */
  disableOnChangeOptimisation?: boolean
```

Reply:
beautiful! thanks